implementations/rust/ockam/ockam_api/src/authenticator/direct/authenticator.rs

```diff
@@ -0,0 +1,132 @@
+use minicbor::Decoder;
+use ockam::identity::utils::now;
+use ockam::identity::{secure_channel_required, TRUST_CONTEXT_ID};
+use ockam::identity::{AttributesEntry, IdentityAttributesReader, IdentityAttributesWriter};
+use ockam::identity::{Identifier, IdentitySecureChannelLocalInfo};
+use ockam_core::api::{Method, Request, Response};
+use ockam_core::compat::sync::Arc;
+use ockam_core::{CowStr, Result, Routed, Worker};
+use ockam_node::Context;
+use std::collections::HashMap;
+use tracing::trace;
+
+use crate::authenticator::direct::types::AddMember;
+
+pub struct DirectAuthenticator {
+    trust_context: String,
+    attributes_writer: Arc<dyn IdentityAttributesWriter>,
+    attributes_reader: Arc<dyn IdentityAttributesReader>,
+}
+
+impl DirectAuthenticator {
+    pub async fn new(
+        trust_context: String,
+        attributes_writer: Arc<dyn IdentityAttributesWriter>,
+        attributes_reader: Arc<dyn IdentityAttributesReader>,
+    ) -> Result<Self> {
+        Ok(Self {
+            trust_context,
+            attributes_writer,
+            attributes_reader,
+        })
+    }
+
+    async fn add_member<'a>(
+        &self,
+        enroller: &Identifier,
+        id: &Identifier,
+        attrs: &HashMap<CowStr<'a>, CowStr<'a>>,
```

Comment:
Do we really benefit from `CowStr` instead of just consuming `Strings` here?

Reply:
We don't, but I just moved that code